source/vir/src/well_formed.rs

```diff
@@ -15,6 +15,30 @@ struct Ctxt {
     pub(crate) dts: HashMap<Path, Datatype>,
 }
 
+fn typ_min_mode(ctxt: &Ctxt, typ: &crate::ast::Typ) -> Mode {
+    use crate::ast::IntRange;
+    match &**typ {
+        TypX::Bool => Mode::Exec,
+        TypX::Int(IntRange::Int | IntRange::Nat) => Mode::Proof,
+        TypX::Int(IntRange::U(_) | IntRange::I(_) | IntRange::USize | IntRange::ISize) => {
+            Mode::Exec
+        }
+        TypX::Tuple(ts) => ts
+            .iter()
+            .map(|t| typ_min_mode(ctxt, t))
+            .reduce(crate::modes::mode_join)
+            .unwrap_or(Mode::Exec),
+        TypX::Lambda(_, _) => Mode::Proof,
```

Comment:
@utaal is this supposed to be `Mode::Spec`, since a `TypX::Lambda` is always the type of an `FnSpec`?

Reply:
Yes! Good catch, thanks!